lib/std/os/windows.zig

```diff
@@ -710,7 +713,10 @@ pub fn ReadLink(dir: ?HANDLE, sub_path_w: []const u16, out_buffer: []u8) ReadLin
     defer CloseHandle(result_handle);
 
     var reparse_buf: [MAXIMUM_REPARSE_DATA_BUFFER_SIZE]u8 = undefined;
-    _ = try DeviceIoControl(result_handle, FSCTL_GET_REPARSE_POINT, null, reparse_buf[0..]);
+    _ = DeviceIoControl(result_handle, FSCTL_GET_REPARSE_POINT, null, reparse_buf[0..]) catch |err| switch (err) {
+        error.AccessDenied => unreachable,
```

Comment:
Wait, I'm confused. I thought we wanted to return `AccessDenied` in case we can't apply reparse point processing data on the target? Why are we panicking instead?

Reply:
My thinking is: creating a symLink may fail, so we have to report that via an error. `AccessDenied` looked like a good candidate, and for now it only models "lack of privilege/developer mode".
But `DeviceIoControl` is used for several different operations, and these two (following a symlink, normalizing a path) likely can't fail for this particular reason. Hence the `unreachable`. However this is an assumption I'm making, please correct me if I'm wrong.

Reply:
Ah, wait, I misread this. Github folded the code so well that I thought this particular catch and panic applied to `CreateSymbolicLink`, and now I see that it applies to `ReadLink`, so apologies for the confusion here! It makes sense to catch `AccessDenied` and panic here IMHO.